flang/lib/Lower/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error -Wno-unused-parameter")
```

Comment:
Any reason to do this change :)? I see failures during build #263

Reply:
Absolutely. It's time to fix all the warnings.